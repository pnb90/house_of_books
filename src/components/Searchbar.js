import React, {useState} from 'react'
import styled from 'styled-components'
import InputBase from '@material-ui/core/InputBase'
import PropTypes from 'prop-types'

const Input = styled(InputBase)`
  background: white;
  border-radius: 5px;
  margin-top: 15px;
  margin-bottom: 15px;
  padding: 5px;
  background: #f1f8e9;
`
function Searchbar(props) {
  const [searchTerm, setSearchTerm] = useState("")

  
  const handleSearchTermsChanges = (event) => { 
    setSearchTerm(event.target.value); 
  }
    
  const handleSubmit = (event) => {
    event.preventDefault()
    props.searchBookCallback( searchTerm.toLowerCase() )
  }

  return(
    <form
      onSubmit = { handleSubmit }
    >
      <Input
        type = "text"
        aria-label = "book searchbar"
        onChange = { handleSearchTermsChanges }
        placeholder = "Search for books"
      />
    </form>
  )
}

Searchbar.propTypes = {
  searchBookCallback: PropTypes.func
}

export default Searchbar