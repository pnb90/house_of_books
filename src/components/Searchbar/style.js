import styled from 'styled-components'
import InputBase from '@material-ui/core/InputBase'

export const Input = styled(InputBase)`
  background: white;
  border-radius: 5px;
  margin-top: 15px;
  margin-bottom: 15px;
  padding: 5px;
  background: #f1f8e9;
`

export const InputLabel = styled.label`
  
`