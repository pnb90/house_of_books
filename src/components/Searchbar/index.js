import React, {useState} from 'react'
import PropTypes from 'prop-types'
import { Input, InputLabel } from './style'

export default function Searchbar(props) {
  const [searchTerm, setSearchTerm] = useState("")
  
  /** 
   * Updates the searchTerm to match the information being typed.
   * 
   * @param {event} - HTML event tracking information in the searchbar
   * 
   * Calls the setSearchTerm hook to update searchTerm upon any changes in the searchbar.
  */
 
  const handleSearchTermsChanges = (event) => { 
    setSearchTerm(event.target.value); 
  }
    
   /** 
   * Handles the submit action once entering the search term
   * 
   * @param {event} - HTML event tracking information in the searchbar
   * 
   * Prevents default behavior of the event, namely reloading the page.
   * Passes the searchTerm back to the parent component via the props.searchBookCallback function after downcasing it. 
   * 
  */
  const handleSubmit = (event) => {
    event.preventDefault()
    props.searchBookCallback( searchTerm.toLowerCase() )
  }

  return(
    <form
      onSubmit = { handleSubmit }
      data-testid = "searchbar testID"
    >   
      <InputLabel htmlFor = "searchInput"> 
        Book Search: 
      </InputLabel> 
      <Input
        id = "searchInput"
        type = "text"
        aria-label = "book searchbar"
        onChange = { handleSearchTermsChanges }
        placeholder = "Search for books"
      />
    </form>
  )
}

Searchbar.propTypes = {
  searchBookCallback: PropTypes.func
}