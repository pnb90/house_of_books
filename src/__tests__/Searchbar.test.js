import React from 'react';
import { mount, shallow, render } from '../enzyme';
import { create } from "react-test-renderer";
import Searchbar from '../components/Searchbar';

describe('Basic test', () => {
  it('basic test', () => {
    expect(2+2).toBe(4)
  })
})

describe('Searchbar', () => {
  it('renders', () => {
    const wrapper = mount(<Searchbar />)
    expect(wrapper).toBeTruthy()
  })
})

describe('Searchbar defaults', () => {
  it('has these defaults', () => {
    const wrapper = mount(<Searchbar />)
    console.log(wrapper.find("Input"))
    expect(wrapper.find("Input"))
  })
})

describe('Searchbar Props', () => {
  it('should do something', () => {
    expect(2+4).toEqual(6)
  })

  it('accept a function as props', () => {
    const wrapper = shallow(
        <Searchbar
          aria-label= "book searchbar"
        />
    )
  })
})