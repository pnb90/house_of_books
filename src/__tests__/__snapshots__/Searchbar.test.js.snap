// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar Component renders the searchbar component without errors 1`] = `
<form
  onSubmit={[Function]}
>
  <Styled(WithStyles(ForwardRef(InputBase)))
    aria-label="book searchbar"
    onChange={[Function]}
    placeholder="Search for books"
    type="text"
  />
</form>
`;
